CONTRIBUTING.rst

```diff
@@ -0,0 +1,109 @@
+======================
+Contributing to DWI_ML
+======================
+
+Welcome to DWI_ML ! We're excited you're here and want to contribute.
+
+This article documents how to contribute improvements to DWI_ML.
+
+The development process
+=======================
+
+When writing code, please pay attention to the following aspects:
+
+Coverage
+--------
+
+We use `pytest`_ to write the tests of the code.
+
+If you are adding code into a module that already has a test file, add
+additional tests into the respective file.
+
```

Comment:
Just opening a discussion.

How are we going to be able to test anything? Tools in dwi_ml are going to be a base ("abstract"), so that people can create their own classes and scripts for their own data. So things will not really be meant to be fully runnable probably (so full scripts?). I guess at some point, the tests are more gonna come from each of our projects, no? I'm not sure we can force people to add tests.

Reply:
I think there is a non-negligible amount of code exclusive to this repository and that could/will eventually be shared among our projects that can be tested. As right now I am not using this code base, I would need to look at specific examples. But the moment when two projects use the same methods is already a powerful reason to think about tests. Even when a method is used by just one project, having a test allows you to be able to reproduce your results or not to break things inadvertently, and be able to demonstrate a few basic things that could be called into question otherwise.

As long as we do not share this viewpoint as owners of the repository, we are not going to be able to enforce writing tests, meaning that we should not merge something that is not tested, including our own code.

A test allows to self-answer questions, questions from other people, and more importantly, to avoid bugs (bugs that may lead to unexpected and incorrect results).

IMHO, we just need to look at the packages we rely on to realize about the importance of this.

Writing tests takes time, and that is a trade-off between other things that need time and writing tests. But advising to do so explicitly is already a step in the right direction.